README.md

```diff
@@ -3,92 +3,117 @@
+   - [nodenv](https://github.com/nodenv/nodenv) - Node version manager
+   - [tfenv](https://github.com/tfutils/tfenv) - Terraform version manager
+   - [terraform-docs](https://terraform-docs.io/user-guide/installation/) - Generate Terraform modules documentation in various formats
+   - [tflint](https://github.com/terraform-linters/tflint) - A Pluggable Terraform Linter
+   - [pre-commit](https://pre-commit.com/) - A framework for managing and maintaining multi-language pre-commit hooks
 
-  ```sh
-  pre-commit install
-  ```
+2. Install `node` at the right version used by this project
 
-## Tasks
+   ```bash
+    cd path/to/project
+    nodenv install
+   ```
 
-Tasks are defined in the `turbo.json` and `package.json` files. To execute a task, just run the command at the project root:
+3. Install `yarn` using [corepack](https://nodejs.org/api/corepack.html) (Node Package Manager version manager, it is distributed with `node`). This step will also install all the required dependencies
 
-```sh
-yarn <cmd>
-```
+> [!IMPORTANT]
+> Yarn uses Plug and Play for dependency management. For more information, see: [Yarn Plug’n’Play](https://yarnpkg.com/features/pnp)
 
-`Turborepo` will execute the task for all the workspaces that declare the same command in their `package.json` file; it also applies caching policies to the command according to the rules defined in `turbo.json`.
+   ```bash
+   corepack enable
+   yarn
+   ```
 
-To define a new task:
+4. Build all the workspaces contained by this repo
+   ```bash
+   yarn build
+   ```
 
-- add the definition to `turbo.json` under `pipeline`;
-- add a script with the same name in `package.json` as `turbo <cmd name>`.
+## Release management
 
-Defined tasks are _lint_, _test_, and _typecheck_.
+We use [changesets](https://github.com/changesets/changesets) to automate package versioning and releases.
 
-## Dependencies
+Each Pull Request that includes changes that require a version bump must include a _changeset file_ that describes the introduced changes.
 
-> [!IMPORTANT]  
-> This project uses Yarn Plug'n'Play as installation strategy for dependencies. [Check out](https://yarnpkg.com/features/pnp) the official Yarn documentation to lean about pnp and its difference from the classic `node_modules` approach.
+To create a _changeset file_ run the following command and follow the instructions.
 
-```sh
-# install all dependencies for the project
-yarn
+```bash
+yarn changeset
+```
 
-# install a dependency to a workspace
-#   (workspace name is the name in the package.json file)
-yarn workspace <workspace name> add <package name>
-yarn workspace <workspace name> add -D <package name>
+## Useful commands
 
-# install a dependency for the monorepo
-#   (ideally a shared dev dependency)
-yarn add -D <package name>
-```
+This project uses `yarn` and `turbo` with workspaces to manage projects and dependencies. Here is a list of useful commands to work in this repo.
```

Comment:
> This project uses `yarn`
Is this going to be replaced?

Reply:
For now we use `yarn` as package manager, `pnpm` is not yet distributed to our consumer base.